rmf_robot_sim_common/src/slotcar_common.cpp

```diff
@@ -266,11 +395,17 @@ std::array<double, 2> SlotcarCommon::calculate_control_signals(
       _nominal_drive_speed,
       _nominal_drive_acceleration, _max_drive_acceleration, dt);
 
-  const double w_target = rmf_plugins_utils::compute_ds(velocities.second,
+  double w_target = rmf_plugins_utils::compute_ds(velocities.second,
       w_robot,
       _nominal_turn_speed,
       _nominal_turn_acceleration, _max_turn_acceleration, dt);
 
+  if (!this->_is_holonomic)
+  {
+    // @note: temporary hack to resolve weird turns
+    w_target = velocities.second;
+  }
+
```

Comment:
this is required. turning hasn't been working correctly (something about bugs with the rate of change). I'm fully expecting to re-look at this code but for now this will have to do